### PR TITLE
Fix emotesets not loading properly

### LINKED ISSUE #2904: Twitchemotes API died making emoteset queries fail to deliver data
**Describe your issue**

Twitchemotes literally just KILLED their entire API about an hour ago. All requests to chatterino-api fail.
This makes it, so all emotesets from Kraken that need to get their emoteset data fetched fail to set extra information about themselves - this results in most (emotesets fetched from IVR API get their emoteset data already, so those are not affected) sub emotes to not show up in emote picker.
You can still tab the Kraken emotes though. It's just emoteset data that's missing.

See related pajbot issue: https://github.com/pajbot/pajbot/issues/1276

**Screenshots**

pajaS

**OS and Chatterino Version**

Arch GNU/Linux 🤓 master

### PR DESCRIPTION
**Pull request by zneix**
Pull request checklist:

- [x] `CHANGELOG.md` was updated, if applicable

# Description

Initial work. It seems to work, but please do not make any reviews yet. WIP

Fixes #2904

### CODE REVIEW

File: src/providers/twitch/TwitchAccount.cpp [hunk 1]
```diff
@@ -508,38 +507,50 @@ void TwitchAccount::loadEmoteSetData(std::shared_ptr<EmoteSet> emoteSet)
...
+                [emoteSetData] {
+                    // epic helix fail2
+                    qCWarning(chatterinoTwitch)
+                        << "Failed to query user by id:" << emoteSetData.ownerId
+                        << emoteSetData.setId;
+                });
+        },
+        [emoteSet] {
+            // epic helix fail
+            qDebug() << "FDM" << emoteSet->key;
```
Comment (fourtf) [inline src/providers/twitch/TwitchAccount.cpp:551]: remove comment and debug

File: src/providers/twitch/TwitchAccount.cpp [hunk 2]
```diff
@@ -508,38 +507,50 @@ void TwitchAccount::loadEmoteSetData(std::shared_ptr<EmoteSet> emoteSet)
...
+    getHelix()->getEmoteSetData(
+        emoteSet->key,
+        [emoteSet](HelixEmoteSetData emoteSetData) {
+            if (emoteSetData.ownerId.isEmpty() ||
+                emoteSetData.setId != emoteSet->key)
             {
-                return Failure;
+                qCWarning(chatterinoTwitch)
+                    << QString("Failed to fetch emoteSetData for %1, assuming "
+                               "Twitch is the onwer")
```
Comment (Mm2PL) [inline src/providers/twitch/TwitchAccount.cpp:518]: You misspelled owner here
```suggestion
                               "Twitch is the owner")
```

File: src/providers/twitch/TwitchAccount.cpp (hunk 1 above)
Comment (zneix) [reply src/providers/twitch/TwitchAccount.cpp:551]: oh sure, I missed this one. Thanks for pointing it out :+1:

File: src/providers/twitch/TwitchAccount.cpp [hunk 3]
```diff
@@ -508,38 +507,50 @@ void TwitchAccount::loadEmoteSetData(std::shared_ptr<EmoteSet> emoteSet)
...
-            qCWarning(chatterinoTwitch)
-                << QString("Error code %1 while loading emote set data for %2")
-                       .arg(result.status())
-                       .arg(emoteSet->key);
-        })
-        .execute();
+            // emote set 0 = global emotes
+            if (emoteSetData.ownerId == "0")
+            {
+                //                emoteSet->channelName = QString();
```
Comment (Mm2PL) [inline src/providers/twitch/TwitchAccount.cpp:531]: Did you intend to leave this commented out or left it by mistake?
```suggestion
```

File: src/providers/twitch/TwitchAccount.cpp [hunk 4]
```diff
@@ -508,38 +507,50 @@ void TwitchAccount::loadEmoteSetData(std::shared_ptr<EmoteSet> emoteSet)
...
+            }
+
+            getHelix()->getUserById(
+                emoteSetData.ownerId,
+                [emoteSet](HelixUser user) {
+                    emoteSet->channelName = user.login;
+                    emoteSet->text = user.displayName;
+                },
+                [emoteSetData] {
+                    // epic helix fail2
```
Comment (Mm2PL) [inline src/providers/twitch/TwitchAccount.cpp:543]: another comment
```suggestion
```

File: src/providers/twitch/api/Helix.cpp [hunk 5]
```diff
@@ -761,6 +761,37 @@ void Helix::getCheermotes(
...
+            }
+
+            HelixEmoteSetData emoteSetData(data.toArray()[0].toObject());
+
+            successCallback(emoteSetData);
+            return Success;
+        })
+        .onError([](NetworkResult result) {
+            qDebug() << "Epix helix fail xd" << result.status();
+        })
```
Comment (Mm2PL) [inline src/providers/twitch/api/Helix.cpp:791]: I believe this qDebug to be unspecific and unhelpful for debugging. Please change or remove it :)

**Review by Mm2PL — CHANGES_REQUESTED** [review]
pajaDent

File: src/providers/twitch/TwitchAccount.cpp (hunk 3 above)
Comment (zneix) [reply src/providers/twitch/TwitchAccount.cpp:531]: I'm not really sure if setting channelName in non-global emotesets from Twitch does not break things (puts them in Global category). I'll leave this commented out for now. Will probably remove it before finally merging into master.

File: src/providers/twitch/api/Helix.cpp (hunk 5 above)
Comment (zneix) [reply src/providers/twitch/api/Helix.cpp:791]: Same deal as with all other debugs; I wrote this code at 4am ok? ;v thanks for pointing out though

**Review by talneoran — APPROVED** [review]
Seems to work 👍 
Only issue is the `qa_TW_Partner` user that twitch seems to use for some of the global emotes, which causes them to show under that user instead of under `Twitch` in the emote popup. 
Could maybe handle this by manually moving emotes from that emote set to the `Twitch` set until this is changed by twitch or another fix is possible.

**Review by Mm2PL — COMMENTED** [review]
Here's how it currently looks. I dislike the `qa_TW_Partner`

**Pull request merged**